code/modules/spells/spell_types/shapeshift.dm

```diff
@@ -22,6 +22,11 @@
 		/mob/living/simple_animal/hostile/construct/armored)
 
 /obj/effect/proc_holder/spell/targeted/shapeshift/cast(list/targets,mob/user = usr)
+	if(src in user.mob_spell_list)
+		user.mob_spell_list.Remove(src)
+		user.mind.AddSpell(src)
```

Comment:
I think you meant to add it to the new body not new mind ?

Reply:
If I add it to the mind then I can stop doing this silly process every time they transform

Reply:
But it will stay if they stop being a vampire ?

Reply:
Oh, nevermind, it's removed already.